azure-tools-common/src/main/java/com/microsoft/azure/common/project/IProject.java

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.azure.common.project;
+
+import java.nio.file.Path;
+import java.util.Collection;
+
+public interface IProject {
+	Path getBaseDirectory();
+
+	Path getJarArtifact();
+
+	Path getBuildDirectory();
+
+	Path getClassesOutputDirectory();
+
+	Collection<Path> getProjectDepencencies();
+
+	boolean isWarProject();
```

Comment:
Change to `Enum getProjectType()`?

Reply:
not convenient when using it(will use getProjectType() == Enum.xxx)